### PR TITLE
parafac analysis added

### PR DESCRIPTION
**Pull request by Farnazmdi**

### CODE REVIEW

**Review by aarmey — CHANGES_REQUESTED** [review]
Can you make a new branch for me and remove the changes that aren't used by the test? I'll merge the test once I figure out how to fix it, and I don't want to merge the other changes at the same time.

**Comment by Farnazmdi** [comment]
Test removed from this branch.

**Comment by Farnazmdi** [comment]
Actually, I will move the tensor function to the test branch.

**Comment by aarmey** [comment]
You can move form_tensor() to the other branch. I will have requests for this branch.

File: de/tensor.py
Comment (aarmey) [inline de/tensor.py:66]: This should use the same settings as I included in the test. For example, line search helps get a solution faster, and you want the initialization to work beyond 6 components.

File: de/tensor.py
Comment (aarmey) [inline de/tensor.py:69]: Why do you have `tfac` and `fac_p`?

Comment (aarmey) [reply de/tensor.py:69]: Also, you should just save the factorization from above, so that you don't have to rerun it here.

**Review by aarmey — CHANGES_REQUESTED** [review]
I'm surprised one component explains so much variance. You're 100% sure you're normalizing across perturbations for each gene?

**Comment by aarmey** [comment]
It's also weird in these figures that components seem to be redundant, but I'm wondering if these plots you've posted are still for random initialization (SVD will always be better).

File: de/tensor.py
Comment (Farnazmdi) [reply de/tensor.py]: that was a typo. I fixed it. 
Saved the factorization in the new commit.

**Comment by Farnazmdi** [comment]
The `factorize()` function is updated as you suggested. Also, I was zscoring columns... I fixed that too.
These figures are updated:
image

Genes:
image

Perturbations:
image

**Review by aarmey — APPROVED** [review]
You definitely don't need to try more than 9 components. After that change this can be merged.

**Pull request merged**
